tools/src/Packages.ts

```diff
@@ -268,13 +271,22 @@ export class Package {
   }
 
   /**
-   * Checks whether package contains some native tests for Android.
+   * Checks whether the package contains native unit tests on the given platform.
    */
   async hasNativeTestsAsync(platform: Platform): Promise<boolean> {
     if (platform === 'android') {
-      return fs.pathExists(path.join(this.path, this.androidSubdirectory, 'src/test'));
+      return (
+        fs.pathExists(path.join(this.path, this.androidSubdirectory, 'src/test')) ||
+        fs.pathExists(path.join(this.path, this.androidSubdirectory, 'src/androidTest'))
+      );
+    }
+    if (platform === 'ios') {
+      return fs
+        .readFileSync(path.join(this.path, this.iosSubdirectory, `${this.podspecName}.podspec`))
+        .toString()
+        .includes('test_spec');
```

Comment:
Is this file guaranteed to exist or do we need to add an existence check like above?

Reply:
good point. with the changes @bbarthec suggested, we are ok now that we check for the `podspecName` in this method.